app/webpacker/controllers/order_info_controller.js

```diff
@@ -0,0 +1,90 @@
+import { Controller } from "stimulus"
+
+export default class extends Controller {
+  static values = { orderNumber: String, shipmentNumber: String }
+  static targets = [ "editor", "editField", "resourceDisplayer" ]
+
+  showEditor() {
+    this.editorTarget.classList.remove('hidden')
+    this.editFieldTarget.focus()
+    this.editFieldTarget.setSelectionRange(-1, -1)
+    this.resourceDisplayerTarget.classList.add('hidden')
+  }
+
+  hideEditor() {
+    this.editorTarget.classList.add('hidden')
+    this.resourceDisplayerTarget.classList.remove('hidden')
+  }
+
+  save() {
+    let data = this.editFieldTarget.value
+    this.makeApiCall(this.params(data))
+  }
+
+  remove() {
+    let data = ''
+    this.confirmRemove(this.params(data))
+  }
+
+  makeApiCall(params) {
+    fetch(this.url, {
+      method: 'PUT',
+      body: JSON.stringify(params),
+      headers: { "Content-type": "application/json; charset=UTF-8" }
+    })
+      .then(function (response) {
+        if (response.ok) {
+          window.location.reload()
+        } else {
+          console.error(response)
+        }
+      })
+  }
+
+  params(data) {
+    if (this.shipmentNumberValue) {
+      return { shipment: { tracking: data } }
+
+    } else {
+      return { note: data }
+
+    }
+  }
+
+  confirmRemove(params) {
+    this.displayAlert((confirmation) => {
+      if (confirmation) {
+        this.makeApiCall(params)
+      }
+    })
+  }
+
+  displayAlert(callback) {
+    let alertBox = document.querySelector('#custom-confirm')
+    let cancelBtn = alertBox.querySelector('button.cancel')
+    let confirmBtn = alertBox.querySelector('button.confirm')
+
+    confirmBtn.addEventListener('click', function() {
+      alertBox.style.display = 'none'
+      callback(true)
+    })
+
+    cancelBtn.addEventListener('click', function() {
+      alertBox.style.display = 'none'
+      callback(false)
+    })
```

Comment:
These event listeners are added each time you click the button. So if I click the delete button, then cancel and then click it again and confirm then the request is sent twice. So I think that this should be initialised at the beginning.

Reply:
At the first place, attaching an event listener to an element should be done in the `initialize()` method. 

But, probably the best way to do this, is to attach the event directly in the DOM through an action (reference: https://stimulus.hotwired.dev/reference/actions)

Reply:
I actually found a bug when I took a look after mkLink commented.
![ezgif com-gif-maker(1)](https://user-images.githubusercontent.com/87677429/183830912-5df38246-5993-45e3-8719-8e56d23f68c0.gif)
If I try to delete the order first and then try to delete `notes/tracknig` the delete box has the contents for deleting the order. Have been fixing this today.

Reply:
And I should be attaching the eventListener through the HTML itself! I'm looking at this today and will try to make the necessary changes by evening. Thanks for your review guys!